packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaForm.tsx

```diff
@@ -0,0 +1,276 @@
+}
+
+export default function TableColumnSchemaForm(
+  props: TableColumnSchemaFormProps,
+) {
+  const { initialData, entityType } = props
+
+  const numColumnModels = useAtomValue(
+    useMemo(() => atom(get => get(tableColumnSchemaFormDataAtom).length), []),
+  )
+
+  const dispatch = useSetAtom(tableColumnSchemaFormDataAtom)
+
+  useEffect(() => {
+    if (initialData) {
+      dispatch({
+        type: 'setValue',
+        value: initialData.map(
+          (cm: ColumnModel): ColumnModelFormData => ({
+            ...cm,
+            jsonSubColumns: cm.jsonSubColumns?.map(
+              (jsc: JsonSubColumnModel): JsonSubColumnModelFormData => ({
+                ...jsc,
+                isSelected: false,
+              }),
+            ),
+            isSelected: false,
+          }),
+        ),
+      })
+    }
+    // Only run on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return (
+    <Box
+      sx={{
+        py: 2.5,
+        borderBottom: '2px solid',
+        borderColor: 'grey.300',
+      }}
+    >
+      <TableColumnSchemaFormActions />
+      <Box
+        display={'grid'}
+        sx={{
+          gridTemplateColumns: COLUMN_SCHEMA_FORM_GRID_TEMPLATE_COLUMNS,
+          py: 2.5,
+          fontWeight: 700,
+        }}
+        gap={'8px'}
+      >
+        <Box>{/* Checkbox */}</Box>
+        <Box sx={{ gridColumn: '2 / span 2' }}>Column Name</Box>
+        <Box>Column Type</Box>
+        <Box>Size</Box>
+        <Box>Max List Length</Box>
+        <Box>Default Value</Box>
+        <Box>Restrict Values</Box>
+        <Box>Facet</Box>
+        {times(numColumnModels, index => {
+          return (
+            <TableColumnSchemaFormRow
+              entityType={entityType}
+              columnModelIndex={index}
+              key={index}
+            />
+          )
+        })}
+      </Box>
+
+      <Button
+        variant={'outlined'}
+        onClick={() => {
+          dispatch({ type: 'appendColumn' })
+        }}
+        startIcon={<AddCircleTwoTone />}
+      >
+        Add Column
+      </Button>
+      {/*  Add / import buttons here  */}
+    </Box>
+  )
+}
+
+function TableColumnSchemaFormActions() {
+  const dispatch = useSetAtom(tableColumnSchemaFormDataAtom)
+
+  const columnModels = useAtomValue(tableColumnSchemaFormDataAtom)
+  const allSelected = getIsAllSelected(columnModels)
+  const numSelected = getNumberOfSelectedItems(columnModels)
+
+  return (
+    <Box display={'flex'} gap={1}>
+      <Button
+        variant={'outlined'}
+        color={'neutral'}
+        onClick={() => {
+          dispatch({ type: 'toggleSelectAll' })
+        }}
+      >
+        {/*
+           MUI Checkbox looks a little different from ours, but it has an indeterminate state
+           TODO: reconcile these differences
+          */}
+        <MUICheckbox
+          size={'small'}
+          checked={allSelected}
+          indeterminate={numSelected > 0 && !allSelected}
+        />
```

Comment:
We have an `.indeterminate-checkbox` class but it doesn't look right (misaligned) so for now just use the MUI checkbox